### PR TITLE
Remove last_seen field from customer

### PR DESCRIPTION
**Pull request by utsav0209**
### Description

Dropped required field last_seen from Customer table and added last_seen_at as required field

### Issue

#720 

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: assets/src/components/customers/CustomersTable.tsx [hunk 1]
```diff
@@ -77,8 +77,10 @@ const CustomersTable = ({
       dataIndex: 'last_seen_at',
       key: 'last_seen_at',
       render: (value: string, record: Customer) => {
-        const {id, pathname, current_url, last_seen} = record;
-        const formatted = dayjs.utc(value || last_seen).format('MMMM DD, YYYY');
+        const {id, pathname, current_url, last_seen_at} = record;
+        const formatted = dayjs
+          .utc(value || last_seen_at)
```
Comment (reichert621) [inline assets/src/components/customers/CustomersTable.tsx:82]: `value` and `last_seen_at` should be the same in this case, so we can just get rid of the `|| last_seen_at` here i believe

File: lib/chat_api/customers/customer.ex [hunk 2]
```diff
@@ -13,7 +13,7 @@ defmodule ChatApi.Customers.Customer do
 
   @type t :: %__MODULE__{
           first_seen: any(),
-          last_seen: any(),
+          last_seen_at: any(),
```
Comment (reichert621) [inline lib/chat_api/customers/customer.ex:16]: i think this is already set below on line 28, so we can just remove this line

**Review by reichert621 — CHANGES_REQUESTED** [review]
thanks for handling this 🎉 just a couple small things but otherwise looks good!

File: priv/repo/migrations/20210414003100_drop_last_seen_customer.exs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+defmodule ChatApi.Repo.Migrations.DropCusotmerLastSeen do
+  use Ecto.Migration
+
+  def change do
+    alter table(:customers) do
+      remove :last_seen
```
Comment (reichert621) [inline priv/repo/migrations/20210414003100_drop_last_seen_customer.exs:6]: we should probably also ensure that any records with a missing `last_seen_at` value get populated with whatever was in `last_seen`

Comment (utsav0209) [reply priv/repo/migrations/20210414003100_drop_last_seen_customer.exs:16]: Oops, I took a pull from master and realized you have already done this!

**Comment by utsav0209** [comment]
@reichert621 any comments on this?

File: priv/repo/migrations/20210417003100_drop_last_seen_customer.exs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+defmodule ChatApi.Repo.Migrations.DropCusotmerLastSeen do
+  use Ecto.Migration
+  import Ecto.Query
+
+  alias ChatApi.Repo
+
```
Comment (reichert621) [inline priv/repo/migrations/20210417003100_drop_last_seen_customer.exs:6]: @utsav0209 mind fixing the warnings above?

Comment (reichert621) [reply priv/repo/migrations/20210417003100_drop_last_seen_customer.exs:6]: i'm also thinking maybe we shouldn't drop this column quite yet -- maybe we can handle this migration in a separate PR to give us time to test that everything is working as expected?

**Review by reichert621 — COMMENTED** [review]
@utsav0209 just a couple small comments!

i'll also want to do some manual testing on this to make sure it doesn't break anything before we deploy to production 👍

File: priv/repo/migrations/20210417003100_drop_last_seen_customer.exs (hunk 4 above)
Comment (utsav0209) [reply priv/repo/migrations/20210417003100_drop_last_seen_customer.exs:6]: Yeah, I think that's a good idea, can put PR later for this migration once we are sure things are working as expected. I will remove the migration from this PR in that case

Comment (utsav0209) [reply priv/repo/migrations/20210417003100_drop_last_seen_customer.exs]: @reichert621 I am also removing the required check for the last_seen field for this to move forward

**Comment by utsav0209** [comment]
@reichert621 any updates on this?

**Review by reichert621 — APPROVED** [review]

**Comment by reichert621** [comment]
thanks for the help on this @utsav0209! ❤️  (sorry for the delay)

**Pull request merged**